neuralpp/symbolic/eliminator.py

```diff
@@ -33,46 +18,91 @@ def _repeat_n(operation: AbelianOperation, expression: Expression, size: Express
     return None  # TODO: expand this
 
 
-def _symbolically_eliminate(operation: AbelianOperation, index: Variable, interval: ClosedInterval, body: Expression,
-                            is_integral: bool) \
-        -> Optional[Expression]:
-    if operation.value == operator.add:
-        if not is_integral:
-            if (result := Eliminator.symbolic_sum(body, index, interval)) is not None:
-                return result
+class Eliminator:
+    def __init__(self, profiler=None):
+        if profiler is None:
+            self.profiler = Profiler()
         else:
-            if (result := Eliminator.symbolic_integral(body, index, interval)) is not None:
-                return result
-
-    if isinstance(body, Constant):
-        # repeat addition: multiplication, repeat multiplication: power
-        return _repeat_n(operation, body, interval.size)
-    return None
-
+            self.profiler = profiler
 
-class Eliminator:
-    @staticmethod
-    def eliminate(operation: AbelianOperation, index: Variable, constraint: Context, body: Expression,
+    def eliminate(self, operation: AbelianOperation, index: Variable, constraint: Context, body: Expression,
                   is_integral: bool, context: Context) -> Expression:
         def eliminate_at_leaves(dotted_interval: DottedIntervals) -> Expression:
             if not isinstance(dotted_interval, DottedIntervals):
                 raise AttributeError("Expect leaves to be DottedIntervals")
-            result = _eliminate_interval(operation, index, dotted_interval.interval, body, is_integral)
+            result = self._eliminate_interval(operation, index, dotted_interval.interval, body, is_integral, context)
             if not dotted_interval.dots:  # empty dots
                 return result
             inverse = operation.inverse(reduce(operation, dotted_interval.dots, operation.identity))
             return operation(result, inverse)
 
         try:
-            conditional_intervals = from_constraint(index, constraint, context, is_integral)
+            # print(f"context: {SymPyExpression.convert(context).sympy_object}")
+            with self.profiler.profile_section("from-constraint"):
+                conditional_intervals = from_constraint(index, constraint, context, is_integral, self.profiler)
             return map_leaves_of_if_then_else(conditional_intervals, eliminate_at_leaves)
         except Exception as exc:
+            raise AttributeError("disable this for now") from exc
             return BasicQuantifierExpression(operation, index, constraint, body, is_integral)
```

Comment:
The `return` should be the default case. But since it's not reachable by the tests and examples we current have, I replaced with raise for debugging purpose (to make sure that it is indeed not reachable).